ops/model.py

```diff
@@ -3489,3 +3525,57 @@ def validate_label_value(cls, label: str, value: str):
         if re.search('[,=]', v) is not None:
             raise ModelError(
                 f'metric label values must not contain "," or "=": {label}={value!r}')
+
+
+class LazyNotice:
+    """Provide lazily-loaded access to a Pebble notice's details.
+
+    The attributes provided by this class are the same as those of
+    :class:`ops.pebble.Notice`, however, the notice details are only fetched
+    from Pebble if necessary (and cached on the instance).
+    """
+
+    id: str
+    user_id: Optional[int]
+    type: Union[pebble.NoticeType, str]
+    key: str
+    first_occurred: datetime.datetime
+    last_occurred: datetime.datetime
+    last_repeated: datetime.datetime
+    occurrences: int
+    last_data: Dict[str, str]
+    repeat_after: Optional[datetime.timedelta]
+    expire_after: Optional[datetime.timedelta]
+
+    def __init__(self, container: Container, id: str, type: str, key: str):
+        self._container = container
+        self.id = id
+        try:
+            self.type = pebble.NoticeType(type)
+        except ValueError:
+            self.type = type
+        self.key = key
+
+        self._notice: Optional[ops.pebble.Notice] = None
+
+    @property
+    def notice(self) -> ops.pebble.Notice:
```

Comment:
What's the use-case for exposing this?

Reply:
That's a fair call. I was thinking in case they wanted a "real" `pebble.Notice` type. But I don't see why you'd need that. Python works more with duck-typing. I've removed it for now, and we can always reconsider if a use case comes up.